designcompose/src/main/java/com/android/designcompose/squoosh/SquooshRoot.kt

```diff
@@ -105,41 +117,35 @@ internal class SquooshAnimationRenderingInfo(
 
 /// Apply layout constraints to a node; this is only used for the root node and gives the DC
 /// layout system the context of what it is being embedded in.
-private fun SquooshResolvedNode.applyLayoutConstraints(constraints: Constraints, density: Density) {
+private fun SquooshResolvedNode.applyLayoutConstraints(constraints: Constraints, density: Float) {
     val rootStyleBuilder = style.asBuilder()
-    val layoutStyleBuilder = style.layout_style.asBuilder()
+    val layoutStyleBuilder = view.style.layout_style.asBuilder()
```

Comment:
What is the difference between `style` and `view.style`?

Reply:
I added a comment -- basically when this method was getting called during intrinsic width/height queries, I noticed that we were returning inconsistent widths/heights over time, because we were slowly mutating `style` (which only gets recreated when we make a new `SquooshResolvedNode` during recomposition -- but layout can happen any number of times before the next recomposition). So this just ensures that we change the view that layout has without corrupting the root node's layout style.